### ISSUE TITLE
Can support .ssh/config HostName alias?

### ISSUE THREAD

**Issue by richex-cn** [Axosoft/vscode-gitlens#1736]
If I configuration remote alias at `.ssh/config`:

```
Host company
    HostName git.company.com
    User git
```

The following VSCode configuration does not take effect:

```
  "gitlens.remotes": [
    {
      "domain": "git.company.com",
      "type": "Gitlab"
    }
  ]
```

![image](https://user-images.githubusercontent.com/25103518/144268430-be6b4a70-e0ce-4df1-bac2-f89676064009.png)

At the same time, I tried to set `domain` as `company`, I got a incorrect URL, for example: `https://company/owner/repo/tree/master`

**Comment by shaperilio** [comment]
The [README](https://github.com/gitkraken/vscode-gitlens#remote-provider-integration-settings-) is not clear on what exactly the settings mean; seems the best hope for documentation is in the discussion in [this issue](https://github.com/gitkraken/vscode-gitlens/issues/1449)

From that, I was able to get this to work as a "Custom" type of remote:

As per @richex-cn's example:

```
    "gitlens.remotes": [
        {
            "regex": "company",
            "type": "Custom",
            "urls": {
                "repository": "https://github.com/${repoBase}/${repoPath}",
                "branches": "https://github.com/${repoBase}/${repoPath}/branches",
                "branch": "https://github.com/${repoBase}/${repoPath}/commits/${branch}",
                "commit": "https://github.com/${repoBase}/${repoPath}/commit/${id}",
                "file": "https://github.com/${repoBase}/${repoPath}?path=${file}${line}",
                "fileInBranch": "https://github.com/${repoBase}/${repoPath}/blob/${branch}/${file}${line}",
                "fileInCommit": "https://github.com/${repoBase}/${repoPath}/blob/${id}/${file}${line}",
                "fileLine": "#L${line}",
                "fileRange": "#L${start}-L${end}"
            }
        }
    ]
```

My guess as to how this all correlates to `git remote -v` output: if you are getting 
```
origin git@company:owner/repo.git (fetch)
origin git@company:owner/repo.git (push)
```
then, for the purposes of custom URLs, this is parsed as
```
origin git@company:${repoBase}/${repoPath}.git
```
Some notes (GitLens v 2022.8.3105):
1. `${repoBase}` and `${repoPath}` appear undocumented; the tooltip on e.g. "repository" shows `${repo}` as the only available replacement.
2. From the [other issue](https://github.com/gitkraken/vscode-gitlens/issues/1449), it seems that "domain" and "regex" are both *search* terms (i.e. mutually exclusive) to match the output of `git remote -v`, what we need for cases where we have an alias defined in `.ssh/config` is to have a search term and a way to replace the host only, so that "type" can be "GitHub" and we can avoid defining all the URLs manually.
3. For what it's worth, "fileRange" should be called "lineRange", in my opinion.

**Comment by huyz** [comment]
@shaperilio Thanks, that did it for me: I was able to use HostName aliases, albeit with all these manual URLs.

The README is misleading in other ways. It makes you believe you have to put more than the domain name in the `regex` field:
```
    "regex": "ssh:\\/\\/(my\\.company\\.com):1234\\/git\\/(.+)",
```
but
```
	"regex": "github-.*",
```
works just fine.

**Comment by shaperilio** [comment]
> ```
> 	"regex": "github-.*",
> ```
> 
> works just fine.

Good point - I have two aliases set on one machine, so if I use your regex, both will work.

I think @richex-cn was right: if you have a "regex" or "domain" entry, and then you say the "type" is, e.g. "Github", then it should take care of defining the URLs for you. It seems that the second it sees "regex" it assumes "type" is "Custom" and you have to define all the URLs yourself.

**Comment by yunsii** [comment]
Yes, regex variables are hard to understand, there is no document about it 🫠
